audeer/core/io.py

```diff
@@ -410,7 +410,15 @@ def list_file_names(
     """List of file names inferred from provided path.
 
     Args:
-        path: path to file, directory or pattern
+        path: path to directory,
+            or path to directory plus file matching pattern,
+            e.g. ``'dir/file.txt'``.
+            If ``recursive`` is ``True``
```

Comment:
To adjust it to the behavior we introduce here:

```suggestion
            If ``file.txt`` exists in ``dir``
            and ``recursive`` is ``True``
```

Reply:
We found a solution to return `file.txt` from sub-folders even if it does not exist in the root folder.